### ISSUE TITLE
PaginationView bug in Modal + NavigationView

### ISSUE THREAD

**Issue by mistralaix** [SwiftUIX/SwiftUIX#174]
Hi,

I have a weird bug when I try to use a PaginationView.
The view architecture looks like this: Modal > NavigationView > ZStack > ScrollView > VStack > PaginationView
I am hiding the navigationBar, I am using these 2 methods on my ZStack:
``` swift
.navigationBarIsTransparent(true)
.hideNavigationBar()
```
On the first load, the navigationBar is hidden and the paginationView is at the top of the modal.
When I try to scroll or to change the page view, the navigationBar is appearing and it is making the scrollview really buggy.
Here's my example:
``` swift
NavigationView {
  ZStack(alignment: .top) {
    ScrollView {
      VStack(spacing: 20) {
                    PaginationView(headerImages, id: \.id, axis: Axis.horizontal, transitionStyle: .scroll, showsIndicators: true) { item in
                        item.image
                            .resizable()
                            .frame(height: 200)
                    }
                    .pageIndicatorAlignment(Alignment.bottom)
                    .frame(height: 200)
      }
    }
  }.navigationBarIsTransparent(true)
   .hideNavigationBar()
}
```
If I try to remove the PaginationView, I don't have the problem anymore.
Thank you for your help!

**Comment by vmanot** [comment]
Does replacing `hideNavigationBar()` with `navigationBarHidden(true)` help fix the issue? Also, what version of iOS are you on?

**Comment by mistralaix** [comment]
I tried replacing it with `navigationBarHidden(true)` and the same thing happened.
I'm on iOS 14.2.

I also tried to use this method `navigationBarIsTransparent(true)` and to set my ZStack with `.edgesIgnoringSafeArea(.all)`
The bug doesn't happen anymore but the navigationBar is not transparent anymore but only for the PaginationView. If I remove the PaginationView and I put a single Image with the same properties, everything works.
